Katas UI string review - QEC (#1695)
microsoft/qsharp: katas/content/qec_shor/index.md

**What you should know to start working on this kata:**

- Basic knowledge of single-qubit and multi-qubit gates
- Basic knowledge of single-qubit and multi-qubit quantum measurements and their effect on quantum systems

@[section]({
    "id": "qec_shor__noise",
    "title": "Noise in Classical and Quantum Systems"
})

> Any quantum system that can carry out quantum computation is inherently noisy. 

Reviewer: Please revert this line change. Quantum systems don't carry out computations themselves, we use them to do that
```suggestion
Any quantum system we can use to carry out quantum computation is inherently noisy. 
```

Author: Neat observation

> Quantum noise can be caused by different physical processes, depending on the type of particle or device you use as a qubit.

Reviewer: I'm not sure I agree with this rewrite. "Your computation" later makes sense, since you have control over your code as a user. "You use a photon as a qubit" doesn't sound as well, since you don't have control over what you use unless it's your very own quantum device that you built yourself.
```suggestion
Quantum noise can be caused by different physical processes, depending on the type of particle or device used as a qubit.
```

From the computation point of view, the presence of noise in the quantum system means that its state can suffer from random errors, and thus end up in a different state from the state you're relying on to do your computations. 
This makes computations unreliable very fast, since the effects of noise accumulate quickly to make computation results effectively random.

Before diving into how to deal with noise in quantum systems, let's take a quick look at how to do that for classical systems.

> The model used for analyzing classical noise is called *binary symmetric channel*. In this model, the classical bits sent through the channel are transmitted correctly with probability $1-p$ and flipped with probability $p$. A flipped bit is called a *bit flip error*.

Reviewer: ```suggestion
The model used for analyzing classical noise is called *binary symmetric channel*. In this model, the classical bits sent through the channel are transmitted correctly with probability $1-p$ and flipped with probability $p$. The error introduced if the bit is flipped is called a *bit flip error*.
```

In this scenario, the information sent through the channel can be protected against the effects of the noise, that is, bit flip errors, using the *repetition code*:

> - On the sender side, for each bit you create and send three copies of itself:

Reviewer: Then it has to be rewritten further, since you don't create bits, you have a message in mind that you want to send, and you create copies
"for each bit you want to send, you create and send three copies of it:"

> This noise model is useful not only for describing noisy communication channels, but also for memory - any classical system that introduces errors in information when it is left on its own, as opposed to systems that introduce errors during information manipulation. Indeed, we assume that no errors are introduced as we copy the bits during encoding or read and compare their values during decoding.

The main idea of quantum error correction is the same as that for classical error correction: encode information with enough redundancy that we can recover the message even from the noisy transmission results.
Dealing with the noise in quantum systems is more challenging than in classical systems, though, due to the limitations imposed by their nature:

> - **No cloning**: You can't replicate the repetition code for quantum systems in a straightforward manner, by duplicating the quantum state several times, since the no-cloning theorem prohibits that.

Reviewer: I'm not sure I agree with this rewrite. Using you here sounds like you, the specific reader, cannot do this (but possibly somebody else can). "We" here emphasizes that nobody can do these things, they're fundamental limitations

Author: The issue with the use of "we" is that it may be interpreted as "we, Microsoft", or "we, humans", as if it was our own limitation. How about this?

```suggestion
- **No cloning**: It's not possible to replicate the repetition code for quantum systems in a straightforward manner, by duplicating the quantum state several times, since the no-cloning theorem prohibits that.
```

Reviewer: Passive voice works for me, yes

- **Observing the system damages information**: Even if you could produce several copies of a quantum state you want to transmit, you wouldn't be able to compare them afterwards without damaging their state.

> - **Errors are continuous**: You need to recover from arbitrary errors that are much more complicated than the bit flip error in classical systems.

Reviewer: Please do a matching rewrite of you-we issue to passive voice here as well. The sentences in the list need to be consistent

    "title": "Bit Flip Code"
})

> Can we reuse the ideas of a classical repetition code for a quantum error correction code? 

Reviewer: Same concern here and for the next two edits: "we" emphasizes that these ideas are the community-wide, not reader-specific

The naive approach to it would be to try and encode a quantum state $\ket{\psi}$ as several copies of itself: 
$\ket{\psi} \rightarrow \ket{\psi} \otimes \ket{\psi} \otimes \ket{\psi}$.
Unfortunately, the no-cloning theorem and the inability to reconstruct a state accurately after measuring it prevent you from doing that.

> You can, however, take a slightly different approach: encode the *basis states* $\ket{0}$ and $\ket{1}$ in repetition code using a unitary transformation, and deduce the effects of this transformation on superposition states based on its linearity:

Reviewer: Please address my earlier comment for this section as well

    "title": "Shor Code"
})

> Can we combine the lessons learned from the bit flip and phase flip error correction codes to be able to detect and correct both $X$ and $Z$ errors? In that case, we'd also be able to handle $Y$ errors as a combination of $X$ and $Z$ errors happening at the same time, and, as a result, we'll be able to detect and correct an arbitrary single-qubit error.

Reviewer: Same we-you concern here
